src/pkg/podman/error_test.go

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2019 – 2021 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package podman
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInternalError(t *testing.T) {
+	type expect struct {
+		IsNil  bool
+		Error  string
+		Search string
+		Wrap   []string
+	}
+
+	testCases := []struct {
+		name   string
+		input  string
+		expect expect
+	}{
+		{
+			name:  "Empty error message",
+			input: "",
+			expect: expect{
+				IsNil: true,
+				Error: "",
+			},
+		},
+		{
+			name:  "Text with no prolog before an error message",
+			input: "There is no prolog before the error message.",
+			expect: expect{
+				Error: "There is no prolog before the error message.",
+			},
+		},
+		{
+			name:  "Text with prolog before an error message",
+			input: "There is the prolog.Error: an error message",
+			expect: expect{
+				Error:  "an error message",
+				Search: "an error message",
+			},
+		},
+		{
+			name:  "Text with prolog before an error message (separated by newline)",
+			input: "There is the prolog.\nError: an error message",
+			expect: expect{
+				Error:  "an error message",
+				Search: "an error message",
+			},
+		},
+		{
+			name:  "Error message with several wrapped errors (prepended with \"Error:\")",
+			input: "Error: level 1: level 2: level 3: level 4",
+			expect: expect{
+				Error:  "level 1: level 2: level 3: level 4",
+				Search: "level 4",
+				Wrap:   []string{"level 1", "level 2", "level 3", "level 4"},
+			},
+		},
+		{
+			name:  "Error message with newline and with errors in \"bullet\" list",
+			input: "Error: foobar:\n  err1\n  err2\n  err3",
+			expect: expect{
+				Error:  "foobar: err1: err2: err3",
+				Search: "err2",
+				Wrap:   []string{"foobar", "err1", "err2", "err3"},
+			},
+		},
+		{
+			name: "Error message from 'podman pull' - unknown registry",
```

Comment:
Let's restrict ourselves to only parsing `podman pull` errors related to authentication. It's going to be a struggle to keep that working over time, so let's not make it worse for no real benefit.